### PR TITLE
Record: SP4096 + Linear LR + Depth Recurrence -- val_bpb=1.0924 (3-seed mean)

### PR DESCRIPTION
**Pull request by dttdrv**
## Summary

**val_bpb: 1.0924** (3-seed mean, std 0.0004) | 15.99 MB artifact | 8xH100 SXM, 600s

No TTT, no SLOT, no n-gram cache, no eval-time adaptation. All four conditions from Issue #1017 satisfied. Evaluation is pure sliding-window at stride=64.

**Improvement over current SOTA (PR #1019, 1.1147 BPB): -0.0223 BPB** (Welch t=-68.85, df=3.84, p << 0.001)

## Results

| Seed | Steps | Pre-quant BPB | Sliding BPB | Artifact |
|------|-------|---------------|-------------|----------|
| 42 | 5,410 | 1.0974 | **1.0927** | 15,987,206 |
| 314 | 5,409 | 1.0977 | **1.0927** | 15,989,311 |
| 999 | 5,408 | 1.0970 | **1.0919** | 15,988,159 |
| **Mean** | | **1.0974** | **1.0924** | |

## Key Change

Linear warmdown to LR=0 (replacing cosine decay to 0.05 floor). This reduces the quantization gap by 61% (0.038 to 0.014 BPB) and pruning by 82%, by allowing weights to fully settle before GPTQ runs.

## Architecture

11-layer, 512-dim SP4096 transformer with:
- MLP 4x with LeakyReLU(0.5)^2
- Depth recurrence on layers 4,5 (from step 3000)
- Parallel residuals from layer 7
- MuonEq-R optimizer (arXiv:2603.28254)
- QK-Gain 5.0, XSA on all 11 layers
- Full Hessian GPTQ int6 per-row, Brotli-10 compression

## Run Command

```bash
DATA_DIR=./data VOCAB_SIZE=4096 SEED=42 \
torchrun --standalone --nproc_per_node=8 train_gpt.py
```

## Test Plan

- [x] 3-seed validation (seeds 42, 314, 999)
- [x] Statistical significance: t=-68.85, p << 0.001 vs PR #1019
- [x] All artifacts under 16,000,000 bytes (max: 15,989,311)
- [x] Training completes within 600s on 8xH100
- [x] No eval-time adaptation of any kind

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — Record: SP4096 + Linear LR + Depth Recurrence -- val_bpb=1.0924 (3-seed mean)

**BPB:** 1.0924 | **Compliance:** LOOKS CLEAN — score-first-per-chunk TTT (legal #1416/#1423 pattern)

**What I found in the code** (head SHA `ea1246055fcb`, file `records/track_10min_16mb/2026-04-06_SP4096_LinearLR_DepthRecurrence_1.0924/train_gpt.py`):

The TTT path at line **1477** implements the score-first-per-chunk pattern: each chunk is scored under `torch.no_grad()` / `inference_mode()` before the `base_model.train()` + SGD adaptation runs on that same chunk, with an `is_last_chunk` guard so the final chunk gets no adaptation pass. This is the structural shape the legal frontier uses (PRs #1416 erichroepke, #1423 aryanbhosale).

Per **Issue #402 and Issue #677**, TTT is legal when each token is scored before the adapter updates on it, and that's what the code does here — chunk `ci` is scored under weights adapted only on chunks `0..ci-1`. No `prequant_ttt_adapt_adamw(val_tokens, ...)` multi-epoch fine-tune, no scored-region SLOT, no target-in-key n-gram cache.

**CPU smoke test (CT2038 proteus-engine, 2026-04-11):** import OK in 4.06s, dim=512, layers=11, vocab=4096, code=90804 B, SMOKE_TEST_PASS

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending standard checks (3-seed validation, 16MB artifact cap, 10-min wallclock on 8×H100 SXM). The compliance picture matches the legal reference frontier and no flags were raised by the classification pass.

*Auto-classification caveat:* this review was drafted by the AST-based classifier against a template derived from manually-reviewed cluster PRs (#1420, #1450, #1487, #1541, #1529, #1533, #1518). If I've misread a subtlety in your eval path — e.g., multi-epoch TTT that I mistook for single-pass, or a target-in-key lookup I missed in a helper function — please flag it and I'll re-run the audit manually.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). CPU smoke test (CT2038 proteus-engine, 2026-04-11): import OK in 4.06s, dim=512, layers=11, vocab=4096, code=90804 B, SMOKE_TEST_PASS. Classification via deterministic AST-based `classify_prs.py` (pattern bank derived from ~65 manually-reviewed PRs earlier in the 2026-04-11 sweep). This review was auto-drafted from a template and spot-checked before posting — if the template misread your code, please call it out so I can iterate the classifier.*
